### DISCUSSION TITLE
Limited pinch zoom in/out.

### DISCUSSION THREAD

**Discussion by ShankarKakumani** [sceneview/sceneform-android#130] [Q&A]
I would like to zoom out till the model is very small. or like to increase size.

 Or is there a way to set the size for models manually?

**Comment by ThomasGorisse** [comment]
Please send your questions in the Discussions/Q&A section and mark it as answered when it's the case

**Comment by RGregat** [comment]
One way is to scale the `Node` where the model is attached to, manually

```java
Vector3 newScale = new Vector3(scaling, scaling, scaling);

if (node != null) {
    node.setLocalScale(newScale);
}
```

You can use this for example to implement a distance dependent scaling of your model. Simple calculate the distance between your camera and the `AnchorNode` where the `Node` is attached to.

To change the scaling with a gesture you have to wrap your `Node` with a `TransformableNode`

```java
node = new Node();
node.setRenderable(renderable);

transformableNode = new TransformableNode(arFragment.getTransformationSystem());
transformableNode.setParent(anchorNode);

node.setParent(transformableNode);
```

I think that should work.
